library/ssl_tls.c

```diff
@@ -4355,7 +4355,9 @@ static int ssl_conf_set_psk_identity( mbedtls_ssl_config *conf,
         return( MBEDTLS_ERR_SSL_BAD_INPUT_DATA );
     }
 
-    conf->psk_identity = mbedtls_calloc( 1, psk_identity_len );
+    /* The PSK Identity can be length zero, but we can't allocate zero bytes,
```

Comment:
Nitpick: You can call `malloc( 0 )`, but IIRC implementations are allowed to return `NULL` which hence cannot be treated as an error anymore.

Reply:
Yup, fixed as per Gilles' solution below.